lib/tasks/allrecipe-init.rake

```diff
@@ -0,0 +1,172 @@
+          #create a new recipe entry
+          new_recipe = author_existence.recipes.new
+
+          new_recipe.outside_profile_id = author_existence.id
+          new_recipe.domain_name_id = 1.to_i
+          new_recipe.url_code = url_array[4]
+          new_recipe.name = url_code_elements[index].text
+          new_recipe.description = descriptions[index].text
+
+          #go to part two of scrapping process to scrape recipe in depth information
+          get_recipe_indepth_info(url, new_recipe)
+        end
+      end
+    rescue OpenURI::HTTPError => e
+      case rescue_me(e)
+      when 1
+        retry
+      when 2
+        next
+      end
+    end
+  end
+  end_time = Time.now
+  puts end_time - start_time
+end
+
+def get_recipe_indepth_info(url, new_recipe)
+  require 'open-uri'
+  require 'nokogiri'
+  require 'unitwise'
+  require 'unitwise/ext'
+
+  @tries = 0
+  begin
+    puts url
+    browser = open(url).read
+    html_doc = Nokogiri::HTML(browser)
+
+    #extract img info
+    new_recipe.img_collection_url = html_doc.css('#lnkOpenCarousel').attr("href").to_s
+    new_recipe.scrape_collection_completed = 0
+
+    #extract time info
+    prepTimeMin = return_total_minutes(html_doc.css('#prepTimeHour em').text.to_i, html_doc.css('#prepMinsSpan em').text.to_i)
+    cookTimeMin = return_total_minutes(html_doc.css('#cookHoursSpan em').text.to_i, html_doc.css('#cookMinsSpan em').text.to_i)
+    readyTimeMin = return_total_minutes(html_doc.css('#totalHoursSpan em').text.to_i, html_doc.css('#totalMinsSpan em').text.to_i)
+    restTimeMin = readyTimeMin - prepTimeMin - cookTimeMin
+    new_recipe.prep_time = prepTimeMin
+    new_recipe.cook_time = cookTimeMin
+    new_recipe.ready_time = readyTimeMin
+    new_recipe.rest_time = restTimeMin
+
+    #extract servings info
+    new_recipe.original_servings_amount = html_doc.css('#lblYield').text[/\d+/]
+    new_recipe.original_servings_type = html_doc.css('#lblYield').text[/\s(.*)/].gsub('- ', '').squish
+
+    #extract instructions info
+    instructions_elements = html_doc.css('div > div > ol > li > span')
+    instructions_array = []
+    instructions_elements.each do |elements|
+      instructions_array.push(elements.text)
+    end
+    new_recipe.instructions = instructions_array
+
+    #save new_recipe
+    new_recipe.save
+
+    #extract ingredients info
+    ingredients = []
+    ingredients_elements = html_doc.css('#liIngredient')
+    ingredients_elements.each do |element|
+      #extract allrecipe_ingredient_code
+      #find or create ar_ingredient_code entry and create ingredient_sub_name
+      ar_ingredient_code = element.attr('data-ingredientid').to_i
+      ar_ingredient_code_existence = find_create_ingredient(ar_ingredient_code)
+
+      #extract ingredient_sub_name and create entry
+      ingredient_sub_name = element.css('#lblIngName').text
+      ar_ingredient_code_existence.ingredient_names.create(ingredient_sub_name: ingredient_sub_name)
```

Comment:
I think if you are using functions to create records, then you should stick with your convention. For example, you used `find_create_ingredient`, but then you used `ar_ingredient_code_existence.ingredient_names.create(ingredient_sub_name: ingredient_sub_name)` immediately afterwards.